cmd/release-controller/sync_verify.go

```diff
@@ -28,19 +28,25 @@ func (c *Controller) ensureVerificationJobs(release *Release, releaseTag *imagev
 				}
 			}
 
+			jobRetries := 0
 			if status, ok := verifyStatus[name]; ok {
+				jobRetries = status.Retries
 				switch status.State {
-				case releaseVerificationStateFailed, releaseVerificationStateSucceeded:
-					// we've already processed this, continue
+				case releaseVerificationStateSucceeded:
 					continue
+				case releaseVerificationStateFailed:
+					jobRetries++
+					if verifyType.Optional || jobRetries > verifyType.MaxRetries {
```

Comment:
I'd expect we'd want tighter anti-flake constraints for release gates than we have in development for flakes like "upgrade gets stuck", "Kube API hangs", "cluster networking melts down", etc.  That's a different class of errors than "failed to provision infra for the test because AWS is having a bad day".  And once we fork off master, like we have for 4.1.z and 4.2.z, we run much fewer tests, so it's already hard to see flaky issues in CI on those releases.  If we get automatic retries, I think we want them to be narrowly around:

> We've already triaged `${FINGERPRINT}` to be caused by `${ISSUE}` that we know is external to our release images.

To do that, we'd want a structure to hold those fingerprints, and we'd want to search through the job output here for the known fingerprints, and `continue` in those cases as well even if we hadn't reached `MaxRetries` yet.

Reply:
I'll open up a separate PR for this change.

Reply:
Optional and MaxRetries should have nothing to do with each other.  If you set MaxRetries, I expect you to retry.  Optional should not be in this condition.